src/sage/groups/matrix_gps/symplectic.py

```diff
@@ -149,7 +172,7 @@ def Sp(n, R, var='a', invariant_form=None):
         raise ValueError('the degree must be even')
 
     if invariant_form is not None:
-        if isinstance(ring, FiniteField):
+        if ring.is_finite(): # Checking is the ring finite?
```

Comment:
```suggestion
        if ring.is_finite():
```
The code clearly says the same thing: DRY principle here.